src/vs/platform/terminal/node/ptyService.ts

```diff
@@ -84,27 +121,302 @@ export class PtyService extends Disposable implements IPtyService {
+		private readonly _onExit: () => void
+	) {
+		super();
+		this._recorder = new TerminalRecorder(cols, rows);
+
+		this._orphanQuestionBarrier = null;
+		this._orphanQuestionReplyTime = 0;
+		this._disconnectRunner1 = this._register(new RunOnceScheduler(() => {
+			this._logService.info(`The reconnection grace time of ${printTime(LocalReconnectConstants.ReconnectionGraceTime)} has expired, so the terminal process with pid ${this._pid} will be shutdown.`);
+			this.shutdown(true);
+		}, LocalReconnectConstants.ReconnectionGraceTime));
+		this._disconnectRunner2 = this._register(new RunOnceScheduler(() => {
+			this._logService.info(`The short reconnection grace time of ${printTime(LocalReconnectConstants.ReconnectionShortGraceTime)} has expired, so the terminal process with pid ${this._pid} will be shutdown.`);
+			this.shutdown(true);
+		}, LocalReconnectConstants.ReconnectionShortGraceTime));
+
+		// TODO: Bring back bufferer
+		// this._bufferer = new TerminalDataBufferer((id, data) => {
+		// 	const ev: IPtyHostProcessDataEvent = {
+		// 		type: 'data',
+		// 		data: data
+		// 	};
+		// 	this._events.fire(ev);
+		// });
+
+		this._register(this._terminalProcess.onProcessReady(e => this.triggerReplay()));
+
+		this._register(this.onOrphanQuestionReply((event: IOrphanQuestionReplyArgs) => {
+			this.orphanQuestionReply();
+		}));
+
+		this._register(this._terminalProcess.onProcessReady(e => this._onProcessReady.fire(e)));
+		this._register(this._terminalProcess.onProcessTitleChanged(e => this._onProcessTitleChanged.fire(e)));
+
+		// Buffer data events to reduce the amount of messages going to the renderer
+		// this._register(this._bufferer.startBuffering(this._persistentTerminalId, this._terminalProcess.onProcessData));
+		this._register(this._terminalProcess.onProcessData(e => {
+			// console.trace(`data ${e}`);
+			this._recorder.recordData(e);
+		}));
+		this._register(this._terminalProcess.onProcessExit(exitCode => {
+			// this._bufferer.stopBuffering(this._persistentTerminalId);
+
+			// const ev: IPtyHostProcessExitEvent = {
+			// 	type: 'exit',
+			// 	exitCode: exitCode
+			// };
+			// this._events.fire(ev);
+
+			// Remove process reference
+			// TODO: Use an event
+			this._onExit();
+		}));
+	}
+	acknowledgeDataEvent(charCount: number): void {
+		return this._terminalProcess.acknowledgeDataEvent(charCount);
+	}
+	getLatency(): Promise<number> {
+		throw new Error('Method not implemented.');
+	}
+
+	public async start(): Promise<ITerminalLaunchError | { persistentTerminalId: number } | undefined> {
+		if (!this._isStarted) {
+			await this._terminalProcess.start();
+			this._isStarted = true;
+		} else {
+			// TODO: Fix me
+			// this._onProcessReady.fire({ pid: (this._terminalProcess as any)._ptyProcess.pid, cwd: '' });
+			this._onProcessReady.fire({ pid: -1, cwd: '' });
+			this._onProcessTitleChanged.fire(this._terminalProcess.currentTitle);
+		}
+		// TODO: Pass back launch error
```

Comment:
This is probably just a matter of verifying that the `start` result was an error and turn that back if so.